src/sql/workbench/browser/modelComponents/declarativeTable.component.ts

```diff
@@ -284,7 +284,9 @@ export default class DeclarativeTableComponent extends ContainerBase<any, azdata
 
 		// check whether the data property is changed before actually setting the properties.
 		const isDataPropertyChanged = !arrayEquals(this.data, finalData ?? [], (a, b) => {
-			return arrayEquals(a, b);
+			return arrayEquals(a, b, (cell1, cell2) => {
```

Comment:
isDataPropertyChanged doesn't work as expected, we have to do deep object comparison. my new use case uncovered it.

Reply:
What new use case is that? The row selection stuff isn't in data so I'm not sure how this is related to this change specifically.

Reply:
updating the selectedRow after initialization, I have observed that when setting the selectedRow property, the dataValues are considered changed. adding the deep equal comparison fixes this issue.